app/Providers/AppServiceProvider.php

```diff
@@ -24,7 +24,7 @@ public function boot(): void
         Gate::define('admin', function (User $user) {
             return $user->username === 'admin';
         });
-        view()->composer('components.language_switcher', function ($view) {
+        view()->composer('components.sections.components.language-switcher', function ($view) {
             $view->with('current_locale', app()->getLocale());
             $view->with('available_locales', config('app.available_locales'));
         });
```

Comment:
Nice! Although view composers can be defined in a separate file: https://laravel.com/docs/11.x/views#view-composers
You can create the file and register it there like so 

` Facades\View::composer('components.language_switcher', ProfileComposer::class);`